src/plugins/camera/camera_impl.cpp

```diff
@@ -1824,6 +1826,8 @@ void CameraImpl::list_photos_async(
                 while (cv_status == std::cv_status::timeout) {
                     request_try_number++;
                     if (request_try_number >= request_try_limit) {
+                        std::lock_guard<std::mutex> status_lock(_status.mutex);
```

Comment:
What about having one lock only? Did you check that `_status.mutex` is now only accessed inside `_capture_request_mutex` or alone? Otherwise this could deadlock.

Reply:
You mean locking `_status.mutex` for the whole `list_photos` requesting? Not sure I want to do that, it could involve thousands of round-trips :confused:.

Now `_captured_request_mutex` is used exactly once, when this thread starts. To prevent this loop from running twice in parallel. And `_status.mutex` is otherwise only used alone.

Not sure how to do otherwise... do you have a suggestion?

Reply:
as long as we don't acquire these mutexs in reverse order anywhere it's fine.

Reply:
Currently we don't. I know it's not ideal, but well...